server/src/main/java/com/flipkart/varadhi/RestVerticle.java

```diff
@@ -51,7 +51,8 @@ public void start(Promise<Void> startPromise) {
                         behaviorProvider.configure(route, def);
                     }
             );
-            route.handler(def.handler());
+            def.preHandlers().forEach(route::handler);
```

Comment:
In what order pre-handlers would be executed ?

let's also add a test before checkin to ensure ordering is as expected.